src/swarm/neo/client/mixins/AllNodesRequestCore.d

```diff
@@ -0,0 +1,648 @@
+            false to abort.
+        Disconnected = type of policy instance to be called when an I/O error
+            causes the connection to the node to break. Should perform behaviour
+            such as notifying the user and resetting any state for the request-
+            on-conn (e.g. any working data).
+        Initialiser = type of policy instance to handle initialisation of the
+            request on this connection (i.e. sending the initial payload and
+            handling the corresponding status code from the node). This policy
+            is expected to provide two functions:
+                * initialise(): handles the actual initialisation process.
+                * reset(): should reset any state required for initialisation to
+                  begin again. (Called after the handler exits, whether due to
+                  an error or not.)
+        Handler = typoe of policy instance to be called to handle the main logic
+            of the request.
+
+*******************************************************************************/
+
+public struct AllNodesRequest ( Request, Connector, Disconnected, Initialiser,
+    Handler )
+{
+    import swarm.neo.client.RequestOnConn;
+    import ocean.io.select.protocol.generic.ErrnoIOException : IOError;
+
+    /// Policy object for handling connection establishment.
+    private Connector connector;
+
+    /// Policy object called upon disconnection.
+    private Disconnected disconnected;
+
+    // Policy object for handling initialisation, once connected.
+    private Initialiser initialiser;
+
+    /// Policy object for handling the main request logic, once initialised.
+    private Handler handler;
+
+    /// Request-on-conn event dispatcher.
+    private RequestOnConn.EventDispatcherAllNodes conn;
+
+    /// Request context.
+    private Request.Context* context;
+
+    /***************************************************************************
+
+        Performs the logic of connection establishment, request initialisation,
+        and request handling, as specified by the policy objects.
+
+    ***************************************************************************/
+
+    public void run ( )
+    {
+        bool reconnect;
+        do
+        {
+            reconnect = false;
+
+            try
+            {
+                if ( !this.connector() )
+                    return; // request aborted
+
+                if ( !this.initialiser.initialise() )
+                    return; // request aborted
+
+                this.handler();
+            }
+            // Only retry in the case of a connection error. Other errors
+            // indicate internal problems and should not be retried.
+            catch ( IOError e )
+            {
+                this.disconnected(e);
+                reconnect = true;
+            }
+            finally
+            {
+                this.initialiser.reset();
```

Comment:
I don't think so. `initialiser.reset` should just reset the initialiser to its initial, default state. If `initialiser.inisialise` hasn't been called, it should be in its default state anyway.

Did you have anything specific in mind?